src/CoreWCF.Primitives/src/CoreWCF/Runtime/Serialization/DataContractEx.cs

```diff
@@ -113,12 +113,12 @@ private static DataContractEx GetDataContractStub(Type clrType)
         private static DataContractEx GetGetOnlyCollectionDataContractStub(int id, RuntimeTypeHandle typeHandle, Type type, int mode)
         {
             var methodInfo = DataContractType.GetMethod("GetGetOnlyCollectionDataContract", BindingFlags.Static | BindingFlags.NonPublic, null,
-                new Type[] { typeof(int), typeof(RuntimeTypeHandle), typeof(Type),  }, null);
+                new Type[] { typeof(int), typeof(RuntimeTypeHandle), typeof(Type), SerializationModeType }, null);
             var idParam = Expression.Parameter(typeof(int), "id");
             var typeHandleParam = Expression.Parameter(typeof(RuntimeTypeHandle), "typeHandle");
             var typeParam = Expression.Parameter(typeof(Type), "type");
             var intModeParam = Expression.Parameter(typeof(int), "mode");
-            var serializationModeParam = Expression.Constant(intModeParam, SerializationModeType); 
+            var serializationModeParam = Expression.Constant(Enum.ToObject(SerializationModeType, mode), SerializationModeType); 
```

Comment:
This isn't quite the correct solution. The first time this method gets called, it will take the passed in mode value and convert it to SerializationModeType and store that as a constant and always use the same value.  This happens to work because we only pass in the one `SerializationModeType` value in a single call point so it happens to always be that one same value, but this is a potential future bug waiting to be tripped up on. The original code was wrong as I misread what the intellisense auto complete was offering. It should be `Expression.Convert` and not `Expression.Constant`.  
```suggestion
            var serializationModeParam = Expression.Convert(intModeParam, SerializationModeType);
```

Reply:
This makes a lot more sense.  I was wondering why it was just using constant.  I've updated the PR.